vnext/Shared/Hasher.cpp

```diff
@@ -0,0 +1,78 @@
+#include "pch.h"
+#include "Hasher.h"
+
+#ifndef NT_SUCCESS
+#define NT_SUCCESS(Status) ((NTSTATUS)(Status) >= 0)
+#endif // NT_SUCCESS
+
+namespace Microsoft::ReactNative {
+
+  SHA256Hasher::SHA256Hasher()
+  {
+    NTSTATUS status;
+    if (!NT_SUCCESS(status = BCryptOpenAlgorithmProvider(&m_hAlg, BCRYPT_SHA256_ALGORITHM, NULL, 0)))
+    {
+      // Failed to load and initialize a CNG provider
+      throw status;
```

Comment:
> throw status; [](http://example.com/codeflow?start=5&length=14)

I wonder if it is a good approach to throw while our code does not expect exceptions and supposed to return nullptr if we cannot read/verify the byte code.
My understanding is that we cannot do any good with the class constructor. 
Should we have a simple function that does the hashing that returns std::optional?